File: BE/models/Product.js
import mongoose from "mongoose";

const ProductSchema = new mongoose.Schema({
    shopId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Shop',
        required: true
    },
    retailerId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    name: {
        type: String,
        required: true
    },
    slug: {
        type: String,
        unique: true
    },
    description: String,
    price: {
        type: Number,
        required: true
    },
    discountPrice: Number,
    stock: {
        type: Number,
        required: true,
        default: 0
    },
    categoryId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Category'
    },
    brandId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Brand'
    },
    images: [String],
    specs: Object,

    ratingAvg: {
        type: Number,
        default: 0
    },
    ratingCount: {
        type: Number,
        default: 0
    },
    status: {
        type: String,
        enum: ['pending', 'approved', 'rejected'],
        default: 'pending'
    },
    isActive: {
        type: Boolean,
        default: true
    }
}, { timestamps: true });

const Product = mongoose.model('Product', ProductSchema);
export default Product;

File: BE/Models/Entities/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BE.Models.Entities
{
    [Table("Products")]
    public class Product
    {
        [Key]
        public long ProductId { get; set; }

        public long ShopId { get; set; }

        [Required]
        [MaxLength(50)]
        public string RetailerId { get; set; }

        [Required]
        [MaxLength(255)]
        public string Name { get; set; }

        public string? Description { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? DiscountPrice { get; set; }

        public int Stock { get; set; }

        public long CategoryId { get; set; }
        public long BrandId { get; set; }

        public string? Image { get; set; }

        public float RatingAvg { get; set; } = 0;
        public long RatingCount { get; set; } = 0;

        [Required]
        [MaxLength(20)]
        public string Status { get; set; } = "active";

        public string Slug { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        // Navigation
        [ForeignKey(nameof(ShopId))]
        public Shop? Shop { get; set; }

        [ForeignKey(nameof(RetailerId))]
        public User? Retailer { get; set; }

        [ForeignKey(nameof(CategoryId))]
        public Category? Category { get; set; }

        [ForeignKey(nameof(BrandId))]
        public Brand? Brand { get; set; }
        
        public ICollection<CartItem> CartItems { get; set; }
        public ICollection<ProductVariant> Variants { get; set; }
        public ICollection<ProductAttribute> ProductAttributes { get; set; } = new List<ProductAttribute>();
        public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
    }
}
